canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,147 @@
+            Future<List<CanaryFragment>> listFragmentResult = executorService.submit(new CanaryListFragmentWorker(streamName, credentialsProvider, dataEndpoint, Regions.fromName(region), fragmentSelector));
+            List<CanaryFragment> newFragmentList = listFragmentResult.get();
+
+            if (newFragmentList.size() > fragmentList.getFragmentList().size())
+            {
+                newFragmentReceived = true;
+            }
+            log.info(MessageFormat.format("newFragmentReceived: {0}", newFragmentReceived));
+
+            fragmentList.setFragmentList(newFragmentList);
+
+            sendFragmentContinuityMetric(newFragmentReceived, streamName, canaryLabel, credentialsProvider, region);
+
+        } catch(Exception e){
+            log.error(e);
+        } 
+    }
+
+    private static void sendFragmentContinuityMetric(Boolean newFragmentReceived, String streamName, String canaryLabel, SystemPropertiesCredentialsProvider credentialsProvider, String region){
+        try{
+            final AmazonCloudWatchAsync cwClient = AmazonCloudWatchAsyncClientBuilder.standard()
+                .withRegion(region)
+                .withCredentials(credentialsProvider)
+                .build();
+
+            final Dimension dimensionPerStream = new Dimension()
+                .withName("StorageWebRTCSDKCanaryStreamName")
+                .withValue(streamName);
+            final Dimension aggregatedDimension = new Dimension()
+                .withName("StorageWebRTCSDKCanaryLabel")
+                .withValue(canaryLabel);
+            List<MetricDatum> datumList = new ArrayList<>();
+            
+            MetricDatum datum = new MetricDatum()
+                .withMetricName("FragmentReceived")
+                .withUnit(StandardUnit.None)
+                .withValue(newFragmentReceived ? 1.0 : 0.0)
+                .withDimensions(dimensionPerStream);
+            datumList.add(datum);
+            MetricDatum aggDatum = new MetricDatum()
+                .withMetricName("FragmentReceived")
+                .withUnit(StandardUnit.None)
+                .withValue(newFragmentReceived ? 1.0 : 0.0)
+                .withDimensions(aggregatedDimension);
+            datumList.add(aggDatum);
+
+            PutMetricDataRequest request = new PutMetricDataRequest()
+                .withNamespace("KinesisVideoSDKCanary")
+                .withMetricData(datumList);
+            cwClient.putMetricDataAsync(request);
+
+        } catch(Exception e){
+            log.error(e);
+        } 
+    }
+
+    public static void main(final String[] args) throws Exception {
+        final String streamName = System.getenv("CANARY_STREAM_NAME");
+        final String canaryLabel = System.getenv("CANARY_LABEL");
+        final String region = System.getenv("AWS_DEFAULT_REGION");
+
+
+        final Integer canaryRunTime = Integer.parseInt(System.getenv("CANARY_DURATION_IN_SECONDS"));
+        log.info("Stream name {}", streamName);
```

Comment:
Exactly. Follow this style throughout the code base to remove the use of MessageFormat.